### PR TITLE
Closes #491 add nil check for hyperlink['id']

### LINKED ISSUE #491: Invalid Links lead to TypeError: no implicit conversion from nil to integer
Thanks for filing an issue. Following these instructions will help us solve your problem sooner.

### Steps to reproduce
Fixture: [roo_bad_link.xlsx](https://github.com/roo-rb/roo/files/2794813/roo_bad_link.xlsx)

1. Create an executable test case for this issue: https://gist.github.com/weilandia/21485b507ce4f08893a2b1b22858f359

### Issue
```
TypeError: no implicit conversion from nil to integer
    /Users/minerva/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/roo-2.8.1/lib/roo/excelx/relationships.rb:7:in `[]'
```

This happens when a sheet has invalid links.
screen shot 2019-01-24 at 22 04 35
screen shot 2019-01-24 at 22 04 40

### System configuration
**Roo version**: 2.8.1

**Ruby version**: 2.5.3

### PR DESCRIPTION
**Pull request by weilandia**
[491-fix-bad-link-bug]

### Summary

Closes #491

### CODE REVIEW

File: lib/roo/excelx/sheet_doc.rb [hunk 1]
```diff
@@ -182,6 +182,8 @@ def extract_hyperlinks(relationships)
         return {} unless (hyperlinks = doc.xpath('/worksheet/hyperlinks/hyperlink'))
 
         hyperlinks.each_with_object({}) do |hyperlink, hash|
+          next unless hyperlink['id']
```
[the remark this answers is not in the document]
Comment (weilandia) [reply lib/roo/excelx/sheet_doc.rb]: Thanks @chopraanmol1 -- That solution works for me!

**Comment by weilandia** [comment]
Thanks for the review @chopraanmol1 -- I made your suggested changes so let me know if anything else needs to be done.

It would be awesome 🙏 to get a version bump for this as we're using a gem that depends on roo and we're getting this error a few times a day.

Thanks again!

File: lib/roo/excelx/relationships.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def to_a
       private
 
       def extract_relationships
-        return [] unless doc_exists?
+        return {} unless doc_exists?
```
Comment (chopraanmol1) [inline lib/roo/excelx/relationships.rb]: Remove the unnecessary trailing space.

**Comment by chopraanmol1** [comment]
@weilandia I've released a new version.

https://rubygems.org/gems/roo/versions/2.8.2

**Pull request merged**